forte/data/base_pack.py

```diff
@@ -630,6 +634,12 @@ def entry_setter(cls: Entry, value: Any, attr_name: str):
                         "which can only accept a mapping whose values are "
                         "`Entry` objects."
                     ) from e
+            elif attr_type[1] == (FNdArray,):
```

Comment:
Would there be a scenario where the type of an attribute is just `FNdArray`? In that case we should check `attr_type[0]` as well.

Reply:
Yeah. It's now updated to `attr_type == (type(None), (FNdArray,))`